Added a labeling tutorial (#35)
juglab/featureforest: docs/tutorial.md

When the training is done, use the "**Predict Slice**" button to make your first segmentation mask. We know the first one is usually not very good 🙁, but don't get disappointed! Good results will come with consistency, so do some more iterations 😊.

![Layer Tools](assets/labeling_2.png){width="300", align=right}
!!! tip
    Add some labels for background and other classes near the boundaries of targeted objects. This way you provide the most distinctive information for the model.

**Don't add too many labels!**

This is an iterative process, so at each iteration, add labels as few as possible, train the RF model, make a prediction, and then repeat the process by adding a bit more labels over the areas where the model made mistakes.  
For labeling, you can use any slices in the training stack. Therefore, it's better to make predictions over the whole stack and add more labels on any slices in which the model has low performance.

!!! tip

>     The default brush size is set to *one*. It is recommended to use very *thin* labels just to avoid annotating a lot of unnecessary pixels. The reason is that usually the neighboring pixels have similar features and this won't provide distinctive information for the training RF model.

Reviewer: ```suggestion
    The default brush size is set to *one*. We recommended using very *thin* labels to avoid annotating unnecessary pixels. Neighboring pixels usually have similar features, which won't provide distinctive information for the training RF model.
```

## What about the post-processing?

> The post-processing is kind of our winning card! Well, since we are using a RF model for pixel classification, the produced masks, usually have holes and fuzzy edges, and post-processing is there to fix that!

Reviewer: ```suggestion
Post-processing is kind of our winning card! Since we are using a RF model for pixel classification, the produced masks usually have holes and fuzzy edges, and post-processing is there to fix that!
```

### Simple post-processing:

> This will use a simple morphology and mean-curvature smoothing to fill holes and smooth the edges to some extent. You can set the number of smoothing iterations.  

Reviewer: ```suggestion
The postprocessing will use a simple morphology and mean-curvature smoothing to fill holes and smooth the edges. You can set the number of smoothing iterations.  
```

The next step is to eliminate small noisy areas that should not be included in the mask (usually from the background class). You can set the area threshold for that, and all small regions with an area below the threshold will be excluded from the mask.

### Post-processing using SAM2
If you select the "**Use SAM Predictor**" option, bounding boxes around the regions of the RF-predicted mask will be used as prompts to SAM2. Then you will get a nice smooth mask as if you provided those prompts to SAM2. Smart, don't you agree?! 😁  

> The "**Use SAM Auto-Segmentation**" option will use SAM2 auto-segmentation prediction, then it will select those regions that have an intersection with the RF mask above the set threshold. We recommend using this option only if you do have *not too many objects* in the image. Otherwise, it will be slow and take some time to make the final mask.

Reviewer: ```suggestion
The "**Use SAM Auto-Segmentation**" option will use SAM2 auto-segmentation prediction and then select regions that intersect with the RF mask above the set threshold. We recommend using this option only if you have *not too many objects* in the image. Otherwise, it will be slow and take some time to make the final mask.
```

!!! info
    - Use the "**Export**" button to save the created mask. This way the labels will be *0* for the background and onwards for other classes. If you save the mask using the napari file menu, labels will be started from *1* and on...

>     - Every time you make a prediction, the result will show up on a new layer unless you unchecked the "**New Layer**" checkbox.

Reviewer: ```suggestion
    - Every time you make a prediction, the result will appear on a new layer unless you uncheck the "**New Layer**" checkbox.
```
